api/service.go

```diff
@@ -288,6 +288,74 @@ func (s *Service) NewCollection(ctx context.Context, req *pb.NewCollectionReques
 	return &pb.NewCollectionReply{}, nil
 }
 
+func (s *Service) UpdateCollection(ctx context.Context, req *pb.UpdateCollectionRequest) (*pb.UpdateCollectionReply, error) {
+	id, err := thread.Cast(req.DbID)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+	token, err := thread.NewTokenFromMD(ctx)
+	if err != nil {
+		return nil, err
+	}
+	d, err := s.getDB(ctx, id, token)
+	if err != nil {
+		return nil, err
+	}
+	cc, err := collectionConfigFromPb(req.Config)
+	if err != nil {
+		return nil, err
+	}
+	if _, err = d.UpdateCollection(cc); err != nil {
+		return nil, err
+	}
+	return &pb.UpdateCollectionReply{}, nil
+}
+
+func (s *Service) DeleteCollection(ctx context.Context, req *pb.DeleteCollectionRequest) (*pb.DeleteCollectionReply, error) {
+	id, err := thread.Cast(req.DbID)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+	token, err := thread.NewTokenFromMD(ctx)
+	if err != nil {
+		return nil, err
+	}
+	d, err := s.getDB(ctx, id, token)
+	if err != nil {
+		return nil, err
+	}
+	if err = d.DeleteCollection(req.Name); err != nil {
+		return nil, err
+	}
+	return &pb.DeleteCollectionReply{}, nil
+}
+
+func (s *Service) GetCollectionIndexes(ctx context.Context, req *pb.GetCollectionIndexesRequest) (*pb.GetCollectionIndexesReply, error) {
+	id, err := thread.Cast(req.DbID)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+	token, err := thread.NewTokenFromMD(ctx)
+	if err != nil {
+		return nil, err
+	}
+	collection, err := s.getCollection(ctx, req.Name, id, token)
+	if err != nil {
+		return nil, err
+	}
+	indexes := collection.GetIndexes()
+	var pbindexes []*pb.Index
```

Comment:
Nit: preallocate slice.

Reply:
`indexes` is a map. I suppose a count could be tracked in addition the for loop that is used to correctly assign indexes. Will do.

Reply:
Hmm, better yet, make `GetIndexes` return a slice.

Reply:
@sanderpick , `len(indexes)` works for maps too.

Reply:
Right, I'm just saying you wouldn't be able to do the normal pattern `indexes[i] = `, since iterating a map doesn't reveal an index. But tracking count would allow that, just a little extra code.

Reply:
Oh, I see what you mean. 👍